### PR TITLE
Fix regex length issue

### PR DESCRIPTION
**Pull request by matt-oconnell**
Allow length to accept min (`length[3,]`), max (`length[,3]`), and range (`length[3,5]`)

### CODE REVIEW

File: util/validate.js [hunk 1]
```diff
@@ -162,10 +162,11 @@ define(function() {
 
         length: {
           test: function(corpus, meta) {
-            var limits = /\[(,?\d+(?:,\d+)?)\]/.exec(meta);
+            var limits = /\[(?:(\d+|\d+,\d+|,\d+|\d+,))\]/.exec(meta);
```
Comment (mikesherov) [inline util/validate.js]: `/\[(?:(\d+|\d*,\d*))\]/`

Comment (weikinhuang) [inline util/validate.js]: invalid regex, this will match `[1]`, `[1,1]`, `[1,]`, `[,1]`, and `[,]`

Comment (mikesherov) [inline util/validate.js]: https://regexper.com/#%2F%5C%5B(%3F%3A(%5Cd%2B%7C%5Cd*%2C%5Cd*))%5C%5D%2F

vs.

https://regexper.com/#%2F%5C%5B(%3F%3A(%5Cd%2B%7C%5Cd%2B%2C%5Cd%2B%7C%2C%5Cd%2B%7C%5Cd%2B%2C))%5C%5D%2F

Comment (weikinhuang) [inline util/validate.js]: i'm not saying yours isn't more efficient, i'm just saying it'll match the case of `[,]`

Comment (mikesherov) [inline util/validate.js]: hmmm, yours is better.

Comment (mikesherov) [inline util/validate.js]: I have a more efficient one, but yours is most clear.

https://regex101.com/r/LRkCF9/1

Comment (weikinhuang) [inline util/validate.js]: 😄

**Pull request merged**